### DISCUSSION TITLE
How to use DeriveIntoActiveModel?

### DISCUSSION THREAD

**Discussion by kun1988** [SeaQL/sea-orm#1871] [Q&A]
```
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(table_name = "password")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: i64,
    pub primary_type: i16,
    pub sub_type: i16,
    pub password: String,
    pub password_salt: String,
    pub fail_count: i16,
    pub fail_time: Option<TimeDateTimeWithTimeZone>,
    pub target_id: i64,
    pub application_id: i64,
    pub created_by: i64,
    pub created_at: TimeDateTimeWithTimeZone,
    pub updated_by: Option<i64>,
    pub updated_at: Option<TimeDateTimeWithTimeZone>,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}

#[derive(DeriveIntoActiveModel)]
pub struct PasswordForUpdate {
    pub id: i64,
    pub password: Option<String>,
    pub fail_count: Option<i16>,
    pub fail_time: Option<Option<TimeDateTimeWithTimeZone>>
}

impl PasswordForUpdate {
    pub(crate) fn new_password(id: i64, password: String) -> Self {
        Self {
            id,
            password: Some(password),
            fail_count: Some(0),
            fail_time: Some(None)
        }
    }

    pub(crate) fn right_password(id: i64) -> Self {
        Self {
            id,
            password: None,
            fail_count: Some(0),
            fail_time: Some(None)
        }
    }

    pub(crate) fn fail_password(id: i64, fail_count: i16) -> Self {
        let mut fail_time = None;
        if fail_count == 1 {
            fail_time = Some(Some(now_utc()));
        }
        Self {
            id,
            password: None,
            fail_count: Some(fail_count),
            fail_time
        }
    }
}
```
-------------------------------------------------------------------------------------------
error[E0277]: the trait bound `ActiveValue<std::string::String>: From<ActiveValue<std::option::Option<std::string::String>>>` is not satisfied
   --> xxxx.rs:163:10
    |
163 | #[derive(DeriveIntoActiveModel)]
    |          ^^^^^^^^^^^^^^^^^^^^^ the trait `From<ActiveValue<std::option::Option<std::string::String>>>` is not implemented for `ActiveValue<std::string::String>`
    |
    = help: the trait `From<ActiveValue<V>>` is implemented for `ActiveValue<std::option::Option<V>>`
    = note: required for `ActiveValue<std::option::Option<std::string::String>>` to implement `Into<ActiveValue<std::string::String>>`
    = note: this error originates in the derive macro `DeriveIntoActiveModel` (in Nightly builds, run with -Z macro-backtrace for more info)

My code is same test_derive_into_active_model_2 in sea_orm::entity::active_model, why has error?

**Comment by dave42w** [comment]
I think that if you are going to add
```
#[derive(DeriveIntoActiveModel)]
```
then you need to remove
```
impl ActiveModelBehavior for ActiveModel {}
```
as that provides the default implementation.

**Reply by kun1988** [reply]
Thanks.  In Docs Info: "Do not delete the Relation enum or ActiveModelBehavior impl block even if they are empty."

**Reply by dave42w** [reply]
> Thanks.  In Docs Info: "Do not delete the Relation enum or ActiveModelBehavior impl block even if they are empty."

Yeah, not very clear is it. I think it means "unless you have written a custom replacement" 🤔🤣
